src/question/order_select/mod.rs

```diff
@@ -0,0 +1,306 @@
+            validate(&self.select.list.choices.choices, self.answers)?;
+        }
+        Ok(ui::Validation::Finish)
+    }
+}
+
+impl Widget for OrderSelectPrompt<'_, '_> {
+    fn render<B: Backend>(
+        &mut self,
+        layout: &mut ui::layout::Layout,
+        backend: &mut B,
+    ) -> io::Result<()> {
+        self.prompt.render(layout, backend)?;
+        self.select.render(layout, backend)
+    }
+
+    fn height(&mut self, layout: &mut ui::layout::Layout) -> u16 {
+        self.prompt.height(layout) + self.select.height(layout) - 1
+    }
+
+    fn cursor_pos(&mut self, layout: ui::layout::Layout) -> (u16, u16) {
+        self.select.cursor_pos(layout)
+    }
+
+    fn handle_key(&mut self, key: ui::events::KeyEvent) -> bool {
+        let prev_at = self.select.get_at();
+    
+        if let ui::events::KeyCode::Char(' ') = key.code {
+            self.select.list.moving = !self.select.list.moving;
+        } else if self.select.handle_key(key) {
+            if self.select.list.moving {
+                let new_at = self.select.get_at();
+    
+                if prev_at < new_at {
+                    self.select.list.choices.choices[prev_at..=new_at].rotate_left(1);
+                } else {
+                    self.select.list.choices.choices[new_at..=prev_at].rotate_right(1);
+                }
+            }
+        } else {
+            return false;
+        }
+    
+        true
+    }
+}
+
+// =============================================================================
+// 
+// =============================================================================
+
+/// The representation of each choice in an [`OrderSelect`].
+///
+/// It is different from [`ListItem`](crate::answer::ListItem) due to an implementation detail.
+#[derive(Debug, Clone, PartialEq)]
+pub struct OrderSelectItem {
+    initial_index: usize,
+    text: Text<String>,
+}
+
+impl OrderSelectItem {
+    /// The index of the choice in the initial list.
+    /// This is not the index rendered aside on the screen.
+    pub fn initial_index(&self) -> usize {
+        self.initial_index
+    }
+
+    /// The content of the choice -- it is what is displayed to the user
+    pub fn text(&self) -> &str {
+        &self.text.text
+    }
+}
+
+impl Widget for OrderSelectItem {
+    fn render<B: Backend>(
+        &mut self,
+        layout: &mut ui::layout::Layout,
+        backend: &mut B,
+    ) -> io::Result<()> {
+        self.text.render(layout, backend)
+    }
+
+    fn height(&mut self, layout: &mut ui::layout::Layout) -> u16 {
+        self.text.height(layout)
+    }
+
+    fn cursor_pos(&mut self, layout: ui::layout::Layout) -> (u16, u16) {
+        self.text.cursor_pos(layout)
+    }
+
+    fn handle_key(&mut self, key: KeyEvent) -> bool {
+        self.text.handle_key(key)
+    }
+}
+
+impl Into<ListItem> for OrderSelectItem {
+    fn into(self) -> ListItem {
+        ListItem { index: self.initial_index, text: self.text.text }
+    }
+}
```

Comment:
It is better to implement `From<OrderSelectItem> for ListItem` instead. See [here](https://rust-lang.github.io/rust-clippy/master/index.html#from_over_into).